types/azure-sb/lib/models/acstokenresult.d.ts

```diff
@@ -14,8 +14,8 @@
 // limitations under the License.
 //
 
-import { Azure } from 'azure-sb';
-import Dictionary = Azure.ServiceBus.Dictionary;
+import { Azure as Az } from 'azure-sb';
+import Dictionary = Az.ServiceBus.Dictionary;
 
 export namespace Azure.ServiceBus.Results {
```

Comment:
Huh. This was _merging_ with `Azure` from `'azure-sb'`, right? It kind of feels like that should still be allowed? The issue that the https://github.com/microsoft/TypeScript/pull/31231 was supposed to fix was when an imported type completely shadowed an exported type. If they merge, and exports from both are accessible, is that a problem?

Reply:
I don't think it *was* merging it. I'll test with an older version of Typescript.

Reply:
Wouldn’t it have to in order to resolve `Dictionary`?

Reply:
So, after testing, they do *not* merge:

```ts
// @Filename: second.ts
export namespace Azure.ServiceBus {
    export interface Dictionary<T> {
        [k: string]: T
    }
}

export namespace Azure.ServiceBus.Results {
    export interface AcsTokenResponse extends Dictionary<string | Dictionary<string>> {
        WrapAccessToken: Dictionary<string>;
        WrapAccessTokenExpiresIn: Dictionary<string>;
    }

    export interface AcsTokenResult {
        parse(acsTokenQueryString: string): AcsTokenResponse;
    }
}

// @Filename: first.ts
import { Azure } from './second';
import Dictionary = Azure.ServiceBus.Dictionary;
export namespace Azure.ServiceBus.Results {
    export interface Id {
        d: Dictionary<number>
    }
}

// @Filename: welove.ts
import { Azure as A } from './first'
import { Azure as B } from './second'
let x: A.ServiceBus.Results./*1*/
let y: B.ServiceBus.Results./*2*/
```

At /*1*/, you see only `Id` and at /*2*/ you see only AcsTokenResponse and Result. And you can't import both with the name Azure.

Reply:
Sorry, was replying.

The old behaviour was that references to `Azure` referred to the imported symbol. The module *exports* a namespace named `Azure` too, but that's (1) different (2) not referred to *inside* the module.

Reply:
🤦‍♂ 

I completely missed that the old file had `import Dictionary = Azure.ServiceBus.Dictionary;`. I only saw it in your added changes, so I thought that previously, `Dictionary` was being resolved against the current namespace scope, which would have implied that it was merged somehow. That was the source of my confusion.

This seems fine. Thanks for investigating and unconfusing me.